From 3da7bba81c983c764063a59a958f302ebeffc385 Mon Sep 17 00:00:00 2001
From: Haiying Wang <Haiying.Wang@freescale.com>
Date: Wed, 29 May 2013 12:21:30 -0400
Subject: [PATCH 023/383] fsl_qman: return ERANGE if querying fq np fields
 fails due to invalid FQID

For FQID error, return -ERANGE.

Signed-off-by: Haiying Wang <Haiying.Wang@freescale.com>
Change-Id: I403a97ef3dc0f8d52e83e3155e6df6e944b51e55
Reviewed-on: http://git.am.freescale.net:8181/2817
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Thorpe Geoff-R01361 <Geoff.Thorpe@freescale.com>
Reviewed-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
[Original patch taken from QorIQ-SDK-V1.5-SOURCE-20131219-yocto.iso]
Signed-off-by: Bin Jiang <bin.jiang@windriver.com>
---
 drivers/staging/fsl_qbman/qman_high.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/drivers/staging/fsl_qbman/qman_high.c b/drivers/staging/fsl_qbman/qman_high.c
index 15567de..4024337 100644
--- a/drivers/staging/fsl_qbman/qman_high.c
+++ b/drivers/staging/fsl_qbman/qman_high.c
@@ -1764,7 +1764,9 @@ int qman_query_fq_np(struct qman_fq *fq, struct qm_mcr_queryfq_np *np)
 		*np = mcr->queryfq_np;
 	PORTAL_IRQ_UNLOCK(p, irqflags);
 	put_affine_portal();
-	if (res != QM_MCR_RESULT_OK)
+	if (res == QM_MCR_RESULT_ERR_FQID)
+		return -ERANGE;
+	else if (res != QM_MCR_RESULT_OK)
 		return -EIO;
 	return 0;
 }
-- 
1.7.5.4

